src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -197,7 +197,7 @@ public function prepare(Request $request): static
         $this->offset = 0;
         $this->maxlen = -1;
 
-        if (false === $fileSize = $this->file->getSize()) {
+        if ($this->tempFileObject || false === $fileSize = $this->file->getSize()) {
```

Comment:
Reading the [user contributed notes](https://www.php.net/manual/en/class.spltempfileobject.php) I wonder if we shouldn't use the `fstat()` method for temp file objects here.

Reply:
Good idea. I have edited the code.